op-proposer/proposer/l2_output_submitter.go

```diff
@@ -375,6 +380,10 @@ func (l *L2OutputSubmitter) SendTransaction(ctx context.Context, tx *types.Trans
 		return err
 	}
 
+	if l.metricsEnabled {
+		// Emit the proposed block Number
```

Comment:
@trianglesphere @sebastianst I am a bit stuck in how we propose the L2 block here. I am trying to use ref_metrics way of emitting L2 block - https://github.com/ethereum-optimism/optimism/pull/5189/files#diff-0ad98c4d91561ec2bf2386d9d3a5d7e766ba866d749163568e04e6e22bcc2f32R98 but I don't have access to l2ref the way we have in batcher [here](https://github.com/ethereum-optimism/optimism/blob/develop/op-batcher/batcher/driver.go#L218) 

Is there a way to get it or should I introduce a custom method just to emit the block number?

Reply:
The main loop that calls `SendTransaction` has the `L2BlockRef` in `output`. Could either pass the `L2BlockRef` here or you call the `RecordL2BlocksProposed` from the main loop after successfully sending.

Reply:
oh ya. I thought of using `rollupClient` to fetch it but I think the 2nd suggestion makes more sense to include it from the caller itself. I will update.

Reply:
removed this as metrics is being called from caller